pkg/cfn/builder/vpc.go

```diff
@@ -128,7 +131,8 @@ func (c *ClusterResourceSet) addNATGateways() error {
 	case api.ClusterDisableNAT:
 		c.noNAT()
 	default:
-		return errors.New(*c.spec.VPC.NAT.Gateway + "is not a valid NAT gateway mode")
+		// TODO validate this before starting to add resources
```

Comment:
Do you want to do it in this PR or separately? I'm okay with doing it separately, but better to open an issue. There are a few things of this nature that could be improved, we also had a more general issue that needs closing on - #488.

Reply:
I was thinking of doing it in a separate PR. The issue you linked to should address this as well so I don't think we need to open an issue specifically for this.